plotter/image.go

```diff
@@ -0,0 +1,130 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image"
+	"math"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Image implements the Plotter interface.
+type Image struct {
+	img            image.Image
+	cols           int
+	rows           int
+	xmin, xmax, dx float64
+	ymin, ymax, dy float64
+}
+
+// NewImage creates a new image plotter.
+// Image will plot img inside the rectangle defined by the
+// (xmin, ymin) and (xmax, ymax) points given in the data space.
+// The img will be scaled to fit inside the rectangle.
+func NewImage(img image.Image, xmin, ymin, xmax, ymax float64) *Image {
+	bounds := img.Bounds()
+	cols := bounds.Dx()
+	rows := bounds.Dy()
+	dx := math.Abs(xmax-xmin) / float64(cols)
+	dy := math.Abs(ymax-ymin) / float64(rows)
+	return &Image{
+		img:  img,
+		cols: cols,
+		rows: rows,
+		xmin: xmin,
+		xmax: xmax,
+		dx:   dx,
+		ymin: ymin,
+		ymax: ymax,
+		dy:   dy,
+	}
+}
+
+// Plot implements the Plot method of the plot.Plotter interface.
+func (img *Image) Plot(c draw.Canvas, p *plot.Plot) {
+	trX, trY := p.Transforms(&c)
+	xmin := trX(img.xmin)
+	ymin := trY(img.ymin)
+	xmax := trX(img.xmax)
+	ymax := trY(img.ymax)
+	rect := vg.Rectangle{
+		Min: vg.Point{xmin, ymin},
+		Max: vg.Point{xmax, ymax},
+	}
+	c.DrawImage(rect, img.img)
+}
+
+// DataRange implements the DataRange method
+// of the plot.DataRanger interface.
+func (img *Image) DataRange() (xmin, xmax, ymin, ymax float64) {
+	xmax = img.xmax
+	xmin = img.xmin
+	ymax = img.ymax
+	ymin = img.ymin
+	return xmin, xmax, ymin, ymax
+}
+
+// GlyphBoxes implements the GlyphBoxes method
+// of the plot.GlyphBoxer interface.
+func (img *Image) GlyphBoxes(plt *plot.Plot) []plot.GlyphBox {
```

Comment:
ok. (I used it for debugging and figured it could be useful for users too. but you're right.)